egs/wsj/s5/steps/libs/nnet3/report/log_parse.py

```diff
@@ -51,36 +97,45 @@ def parse_progress_logs_for_nonlinearity_stats(exp_dir):
     parse_regex = re.compile(
         ".*progress.([0-9]+).log:component name=(.+) "
         "type=(.*)Component,.*"
-        "value-avg=\[.*mean=([0-9\.\-e]+), stddev=([0-9\.e\-]+)\].*"
-        "deriv-avg=\[.*mean=([0-9\.\-e]+), stddev=([0-9\.e\-]+)\]")
+        "value-avg=\[.*=\((.+)\), mean=([0-9\.\-e]+), stddev=([0-9\.e\-]+)\].*"
+        "deriv-avg=\[.*=\((.+)\), mean=([0-9\.\-e]+), stddev=([0-9\.e\-]+)\]")
 
     for line in progress_log_lines.split("\n"):
         mat_obj = parse_regex.search(line)
         if mat_obj is None:
             continue
-        # groups = ('9', 'Lstm3_i', 'Sigmoid', '0.502', '0.23',
-        # '0.134', '0.0397')
+        # groups = ('9', 'Lstm3_i', 'Sigmoid', '0.05...0.99', '0.502', '0.23',
+        # '0.009...0.21', '0.134', '0.0397')
         groups = mat_obj.groups()
-        iteration = int(groups[0])
-        component_name = groups[1]
         component_type = groups[2]
-        value_mean = float(groups[3])
-        value_stddev = float(groups[4])
-        deriv_mean = float(groups[5])
-        deriv_stddev = float(groups[6])
-        try:
-            stats_per_component_per_iter[component_name][
-                'stats'][iteration] = [value_mean, value_stddev,
-                                       deriv_mean, deriv_stddev]
-        except KeyError:
-            stats_per_component_per_iter[component_name] = {}
-            stats_per_component_per_iter[component_name][
-                'type'] = component_type
-            stats_per_component_per_iter[component_name]['stats'] = {}
-            stats_per_component_per_iter[component_name][
-                'stats'][iteration] = [value_mean, value_stddev,
-                                       deriv_mean, deriv_stddev]
-
+        if component_type == 'LstmNonlinearity':
+            parse_regex_lstmp = re.compile(
```

Comment:
I think this function is still big. I recommend moving this code to a new function.